elasticsearch-model/lib/elasticsearch/model/indexing.rb

```diff
@@ -153,7 +153,24 @@ def mapping(options={}, &block)
         #
         #     # => {:index=>{:number_of_shards=>1}}
         #
+        # You can specify a YAML file with settings
+        #
+        # @example Define index settings from YAML file
+        #
+        #     # config/elasticsearch/articles.yml:
+        #     #
+        #     # index:
+        #     #   number_of_shards: 1
+        #     #
+        #
+        #     Article.settings "config/elasticsearch/articles.yml"
+        #
+        #     Article.settings.to_hash
+        #
+        #     # => { "index" => { "number_of_shards" => 1 } }
+        #
         def settings(settings={}, &block)
+          settings = YAML.load_file(settings) if settings.is_a?(String)
```

Comment:
I think we can improve this significantly by checking whether the argument is an `IO` object, ie. when it responds to `read`. This would not only make testing easier, but would allow people to use the feature in a much more interesting way -- the settings then don't have to come from a physical file on disk, but from Elasticsearch itself, from an API, wrapped in `StringIO`, etc etc etc. What do you think?

I'd also like to allow both `.yml` and `.json` files here, just to have parity with Elasticsearch itself (we can always check the extension when the argument is a physical file with a name).

What do you think? Is it something you'd like to take on or should I take it?